spec/features/cart/checkout_spec.rb

```diff
@@ -27,6 +29,54 @@
 
       expect(current_path).to eq("/orders/new")
     end
+
+    it "requires a visitor to login before checking out" do
+      visit "/cart"
+
+      expect(page).to have_content("You must Register or Log In to finish checking out.")
+      expect(page).to have_link("Register")
+      expect(page).to have_link("Log In")
+    end
+
+    it "has a Register link which links to registration page" do
+      visit "/cart"
+
+      # Without the below, it was giving an Ambiguous Match error because there are two "Register"'s on the page if they're not logged in.  One in the nav, one in the message below.  When I put the ! in front of the within, with the hope that that means "not within", the test does pass ... but I am not sure it's actually working, or if the test just doesn't understand what that is trying to do and is skipping it entirely.
+      # The reason I think it's working is because line 50 *is* passing, which wouldn't make any sense if line 47 wasn't working.  Something to think on...
```

Comment:
WHOA - this seems like a pretty big view issue - Would renaming one of the links a little differently help give our users easier access to register or is it literally just that there are 2 places to register on this one web page?

Reply:
I will probably pull this down and check OR do a code review with you to answer this question